src/briefcase/integrations/android_sdk.py

```diff
@@ -126,6 +126,64 @@ def verify_android_sdk(command):
     return sdk
 
 
+def _get_choices_and_names(running_devices, all_emulators, avd_name_by_device_id):
+    '''Input:
+
+    - `running_devices`: Dict from ADB device ID to dict with 'model' property.
+    - `all_emulators`: List of emulator device names.
+    - `avd_name_by_device_id`: A dict from ADB device ID to AVD name.
+
+    Output:
+
+    Tuple of `choice_ids`, `full_names`.
+
+    `choice_ids` is a list of all running devices, followed by all non-running
+    AVD names prefixed with `@`.
+
+    `full_names` is dict from string (device ID *or* AVD) to a formatted description
+    of the device.
+    '''
+    choice_ids = []
+    full_names = {}
+
+    running_avds = set()  # Used to avoid adding an AVD twice to the list of choices.
+
+    # Handle devices that are running.
+    for device_id, device_info in sorted(running_devices.items(), key=lambda d: d[1]["name"]):
+        model = device_info["name"]
+        if avd_name_by_device_id.get(device_id):
```

Comment:
Not a fan of the "Null as false" use here. This is either an "if device_id in avd_name_by_device_id", or a try/catch.